programs/governance-rewards/src/instructions/reclaim/reclaim_funds.rs

```diff
@@ -0,0 +1,81 @@
+use anchor_lang::prelude::*;
+use anchor_spl::token::{self, Token, TokenAccount};
+
+use crate::{
+    distribution_payout_seeds, error::GovernanceRewardsError, state::distribution::Distribution,
+};
+
+#[derive(Accounts)]
+pub struct ReclaimFunds<'info> {
+    admin: Signer<'info>,
+
+    #[account(mut)]
+    from: Account<'info, TokenAccount>,
+
+    #[account(mut)]
+    to: Account<'info, TokenAccount>,
+
+    #[account(mut)]
+    distribution: Box<Account<'info, Distribution>>,
+
+    /// CHECK: Not read
+    #[account(seeds = [b"payout authority".as_ref(), distribution.key().as_ref()], bump)]
+    pub payout_authority: AccountInfo<'info>,
+
+    token_program: Program<'info, Token>,
+}
+
+impl<'info> ReclaimFunds<'info> {
+    pub fn transfer_context(&self) -> CpiContext<'_, '_, '_, 'info, token::Transfer<'info>> {
+        let token_program = self.token_program.to_account_info();
+
+        let to = self.to.to_account_info();
+
+        let accounts = token::Transfer {
+            from: self.from.to_account_info(),
+            to,
+            authority: self.payout_authority.to_account_info(),
+        };
+        CpiContext::new(token_program, accounts)
+    }
+}
+
+pub fn reclaim_funds(ctx: Context<ReclaimFunds>) -> Result<()> {
+    require!(
+        ctx.accounts.distribution.admin == ctx.accounts.admin.key(),
+        GovernanceRewardsError::AdminOnly
```

Comment:
optionally you could write this as a `has_one = admin` constraint on the `distribution` account

Reply:
I'm open to changing my mind on this but my feeling is that it's confusing to have calling conditions spread out between the account declaration and require! statements.

Reply:
I'd only argue weakly/from opinion here:
- they are already split: this is a signer / this is a tokenaccount / this has these seeds are all constraints
- I think it's the more anchor-y style to declare as much account validation as possible in the accounts struct
- could also move the `!ctx.accounts.distribution.can_register()` into a `constraint = ...`

Reply:
Yeah, ok. I moved the has_one constraint in. This is really more of a beef with anchor that I have.